app/src/main/java/com/github/dedis/student20_pop/utility/security/Hash.java

```diff
@@ -14,26 +15,53 @@
 
     public static final String TAG = Hash.class.getSimpleName();
 
+    private static final char DELIMITER = '\"';
+
+    /**
+     * Hash some objects using SHA256.
+     * Concatenate the object's string representation following the protocol's directive.
+     * Then hash the obtained string
+     *
+     * @param data : the objects to hash
+     * @return the hashed data or null if failed to hash
+     *
+     * @throws IllegalArgumentException if the data is null
+     */
+    public static String hash(Object... data) {
+        if (data == null)
+            throw new IllegalArgumentException("Can't hash a null data");
+
+        StringJoiner joiner = new StringJoiner(",", "[","]");
+        for(Object elem : data)
+            joiner.add(DELIMITER + sanitize(elem.toString()) + DELIMITER);
+
+        return hash(joiner.toString());
+    }
+
+    private static String sanitize(String input) {
```

Comment:
It could be good to rename the method `esc` to follow the specification (#133)

Reply:
Done